code-challenges.test.js

```diff
@@ -14,39 +17,176 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
+describe("codedMessage", () => {
+    it("takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.", () => {
+
+        const secretCodeWord1 = "Lackadaisical";
+        const secretCodeWord2 = "Gobbledygook";
+        const secretCodeWord3 = "Eccentric";
+
+        expect(codedMessage(secretCodeWord1)).toEqual(
+            ("L4ck4d41s1c4l")
+        );
+        expect(codedMessage(secretCodeWord2)).toEqual(
+            ("G0bbl3dyg00k")
+        );
+        expect(codedMessage(secretCodeWord3)).toEqual(
+            ("3cc3ntr1c")
+        );
+    })
+  })
+// Good Fail: ReferenceError: codedMessage is not defined
 
 // b) Create the function that makes the test pass.
 
+// * Resouces used *
+// https://www.tutorialrepublic.com/faq/how-to-replace-character-inside-a-string-in-javascript.php
+    // info on how to replace characters in a string using replace.string method 
+        // syntax: string.replace(/character to be replaced here/, "desired character here")
+// https://thispointer.com/javascript-replace-multiple-characters-in-string/
+    // info on how to add multiple replace methods, we can chain them with .replace after each set
+
+// INPUT:
+    // const secretCodeWord1 = "Lackadaisical";
+    // const secretCodeWord2 = "Gobbledygook";
+    // const secretCodeWord3 = "Eccentric";
+// OUTPUT:
+    // L4ck4d41s1c4l
+    // G0bbl3dyg00k
+    // 3cc3ntr1c
+
+// First I'll create a function codedMessage calling on str
+const codedMessage = (str) => {
+    // I'll make a new let for a new string that we will use to replace the characters
+    let newStr =
+    // To replace the first letter I'll use the string.replace method:
+    // This only replaces the first instance of 'a' so to replace it globally we just add a 'g' outside of the /a/
+    str.replace(/a/g, "4")
+    // Repeat the process for every letter to be replaced
+    .replace(/e/g, "3")
+    // Include capital E to be replaced as we have 'Eccentric'
+    .replace(/E/g, "3")
```

Comment:
I would recommend doing this for all the vowels, however, you can combine it with the lowercase letter as well.